db/tikvgc.go

```diff
@@ -22,10 +22,11 @@ const (
 
 // StartTikvGC start tikv gcwork
 func StartTikvGC(db *DB, tikvCfg *conf.TikvGC) {
-	ticker := time.Tick(tikvCfg.Interval)
```

Comment:
我注意到所有的 `time.Tick` 都换成了 `time.NewTicker`. 目的是什么呢? 这个函数是否发生了变化?

> Tick is a convenience wrapper for NewTicker providing access to the ticking channel only. While Tick is useful for clients that have no need to shut down the Ticker

似乎 `time.NewTicker` 只是增加了 stop 功能 , 而这个 tick 可以不用 stop.

Reply:
这里都换成了time.NewTicker 主要是考虑对象可以重复使用,并且在方法退出的时候能够主动关闭channel,同时也是为了gc/expire/zt/tikvgc 都保持一样的风格

Reply:
可是你这里并没有重用, 而且也没必要关闭.

如果是为了保持风格. 我觉得 ok. 不过这里的 defer 确实没有效果

Reply:
> Stop turns off a ticker. After Stop, no more ticks will be sent. Stop does not close the channel, to prevent a concurrent goroutine reading from the channel from seeing an erroneous "tick".

这个地方也不会关闭 channel

跟 timer.Stop 是一样的

Reply:
如果是为了统一风格, 我觉得 ok, 

不过这里的 defer 确实没有达到你预期的效果, 而且没有重用

Reply:
对的,stop 并没有关闭channel,这个地方可能跟我的预期不太一致

但是除了统一风格外,其实还是因为对非法参数的处理方式不同,我更倾向于当检查到参数非法的时候直接panic 而不是直接返回一个nil
NewTick
>NewTicker returns a new Ticker containing a channel that will send the time with a period specified by the duration argument. It adjusts the intervals or drops ticks to make up for slow receivers. The duration d must be greater than zero; if not, NewTicker will panic. Stop the ticker to release associated resources.

Tick
> Tick is a convenience wrapper for NewTicker providing access to the ticking channel only. While Tick is useful for clients that have no need to shut down the Ticker, be aware that without a way to shut it down the underlying Ticker cannot be recovered by the garbage collector; it "leaks". Unlike NewTicker, Tick will return nil if d <= 0.

虽然在现有的场景中,work 应该是不会停止的,但是我觉的增加defer 是不是语意更加完整一点,

Reply:
赞

Reply:
原来会 leak, 感觉这个有点可怕, 学到了